packages/commonwealth/scripts/add-component-showcase.ts

```diff
@@ -0,0 +1,99 @@
+import fs from 'fs';
+import path from 'path';
+import process from 'process';
+import readlineSync from 'readline-sync';
+
+const ComponentType = {
+  Foundations: 'Foundations',
+  Components: 'Components',
+};
+
+const PATH = 'client/scripts/views/pages/ComponentsShowcase';
+
+const createComponentShowcaseFile = (displayName: string) => {
+  const filePath = path.join(
+    process.cwd(),
+    `${PATH}/components/${displayName}.showcase.tsx`,
+  );
+
+  const reactComponentBoilerplate = `import React from 'react';
+
+const ${displayName}Showcase = () => {
+  return <>{/* ${displayName} component goes here */}</>;
+};
+
+export default ${displayName}Showcase;
+`;
+
+  fs.writeFileSync(filePath, reactComponentBoilerplate);
+};
+
+const modifyComponentsListFile = (
+  displayName: string,
+  componentType: string,
+): void => {
+  const filePath = path.join(process.cwd(), `${PATH}/componentsList.ts`);
+  const fileContent = fs.readFileSync(filePath, 'utf-8');
+
+  const newImportStatement = `import ${displayName}Showcase from 'views/pages/ComponentsShowcase/components/${displayName}.showcase';\n`;
+  const contentWithImport = newImportStatement + fileContent;
+  const modifiedContent = contentWithImport
+    // Adds type to ComponentPageName
+    .replace(
+      /export const ComponentPageName = {([^}]+)}/,
+      `export const ComponentPageName = {$1  ${displayName}: '${displayName}',\n}`,
+    )
+    // Add object to componentItems array
+    .replace(
+      /export const componentItems = \[([^]+)\];/,
+      `export const componentItems = [$1  {\n    ComponentPage: ${displayName}Showcase,\n    displayName: ComponentPageName.${displayName},\n    type: ComponentType.${componentType},\n  },\n]`,
```

Comment:
So I think we need to add a semicolon at the end of the second argument of this replace string

Reply:
nice catch, your suggestion works!